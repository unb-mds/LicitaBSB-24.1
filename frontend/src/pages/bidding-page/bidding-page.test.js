import { screen, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import BiddingPage from '.'
import { MemoryRouter, Route, Routes } from 'react-router-dom'

describe("Deve renderizar as seguintes informações da licitação: ", () => {
    test("tipo da licitação e órgão.", () => {
        const mockId = '10101010'
        render(
            <MemoryRouter initialEntries={[`/licitacoes/${mockId}`]}>
                <Routes>
                    <Route path='/licitacoes/:id' element={<BiddingPage />} />
                </Routes>
            </MemoryRouter>
        )
        const info = screen.getAllByRole('heading')
        expect(info).toHaveLength(2)
    })
    test("lista de links para compartilhamento.", () => {
        const mockId = '10101010'
        render(
            <MemoryRouter initialEntries={[`/licitacoes/${mockId}`]}>
                <Routes>
                    <Route path='/licitacoes/:id' element={<BiddingPage />} />
                </Routes>
            </MemoryRouter>
        )
        const listaDeLinks = screen.getAllByTestId('role-link-id')
        expect(listaDeLinks).toHaveLength(3)
    })
    test("lista de imagens referentes aos links de compartilamento", () => {
        const mockId = '10101010'
        render(
            <MemoryRouter initialEntries={[`/licitacoes/${mockId}`]}>
                <Routes>
                    <Route path='/licitacoes/:id' element={<BiddingPage />} />
                </Routes>
            </MemoryRouter>
        )
        const listaDeLinks = screen.getAllByRole('img')
        expect(listaDeLinks).toHaveLength(4)
    })
    test("objeto da licitação", () => {
        const mockId = '10101010'
        render(
            <MemoryRouter initialEntries={[`/licitacoes/${mockId}`]}>
                <Routes>
                    <Route path='/licitacoes/:id' element={<BiddingPage />} />
                </Routes>
            </MemoryRouter>
        )
        const objeto = screen.getByTestId('objeto-test-id')
        expect(objeto).toBeInTheDocument()
    })
})

test("Deve renderizar o snapshot da lista de links para compartilhamento", () => {
    const mockId = '10101010'
    render(
        <MemoryRouter initialEntries={[`/licitacoes/${mockId}`]}>
            <Routes>
                <Route path='/licitacoes/:id' element={<BiddingPage />} />
            </Routes>
        </MemoryRouter>
    )
    const listaDeLinks = screen.getAllByTestId('role-link-id')
    expect(listaDeLinks).toMatchSnapshot()
})