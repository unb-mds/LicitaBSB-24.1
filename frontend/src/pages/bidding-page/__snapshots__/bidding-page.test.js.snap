// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o snapshot da lista de links para compartilhamento 1`] = `
Array [
  <a
    data-testid="role-link-id"
    href=""
  >
    <img
      src="twitter.svg"
    />
  </a>,
  <a
    data-testid="role-link-id"
    href=""
  >
    <img
      src="facebook.svg"
    />
  </a>,
  <a
    data-testid="role-link-id"
    href=""
  >
    <img
      src="google.svg"
    />
  </a>,
]
`;
